tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/CustomizedCodeUpdateTool/TspClientUpdateTool.cs

```diff
@@ -0,0 +1,639 @@
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "Map MCP failed: {sessionId}", sessionId);
+            SetFailure();
+            return new TspClientUpdateResponse { ResponseError = ex.Message };
+        }
+    }
+
+    [McpServerTool(Name = "tsp_update_merge"), Description("Apply direct merges for unaffected customization files")] 
+    public async Task<TspClientUpdateResponse> Merge(string sessionId, CancellationToken ct = default)
+    {
+        try
+        {
+            ct.ThrowIfCancellationRequested();
+            return await DoMerge(sessionId, ct);
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "Merge MCP failed: {sessionId}", sessionId);
+            SetFailure();
+            return new TspClientUpdateResponse { ResponseError = ex.Message };
+        }
+    }
+
+    [McpServerTool(Name = "tsp_update_propose"), Description("Propose patch diffs for impacted customization files")] 
+    public async Task<TspClientUpdateResponse> Propose(string sessionId, string? filesCsv = null, CancellationToken ct = default)
+    {
+        try
+        {
+            ct.ThrowIfCancellationRequested();
+            return await DoPropose(sessionId, filesCsv, ct);
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "Propose MCP failed: {sessionId}", sessionId);
+            SetFailure();
+            return new TspClientUpdateResponse { ResponseError = ex.Message };
+        }
+    }
+
+    [McpServerTool(Name = "tsp_update_apply"), Description("Apply previously proposed patches (supports dry-run)")] 
+    public async Task<TspClientUpdateResponse> Apply(string sessionId, bool dryRun = false, CancellationToken ct = default)
+    {
+        try
+        {
+            ct.ThrowIfCancellationRequested();
+            return await DoApply(sessionId, dryRun, ct);
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "Apply MCP failed: {sessionId}", sessionId);
+            SetFailure();
+            return new TspClientUpdateResponse { ResponseError = ex.Message };
+        }
+    }
+
+    [McpServerTool(Name = "tsp_update_status"), Description("Return current session status and counts")] 
+    public async Task<TspClientUpdateResponse> Status(string sessionId, CancellationToken ct = default)
+    {
+        try
+        {
+            ct.ThrowIfCancellationRequested();
+            return await DoStatus(sessionId, ct);
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "Status MCP failed: {sessionId}", sessionId);
+            SetFailure();
+            return new TspClientUpdateResponse { ResponseError = ex.Message };
+        }
+    }
+
+    // --------------- Internal Stage Methods ---------------
+    private Task<TspClientUpdateResponse> DoRegenerate(string specPath, string? sessionId, string? newGeneratedPath, CancellationToken ct)
```

Comment:
I see a method name pattern starting with prefix "Do" but that doesn't seems like usual .NET style. For e.g. `Regenerate` is good enough here. 
@benbp @richardpark-msft  Please share your thoughts on this.